assets/pages/regular-contributions/components/regularContributionsPayment.jsx

```diff
@@ -79,7 +79,21 @@ function RegularContributionsPayment(props: PropTypes, context) {
   const isDirectDebitEnable = window.guardian && window.guardian.directDebitEnable;
 
   if (props.country === 'GB' && isDirectDebitEnable) {
-    directDebitButton = <DirectDebitPopUpButton />;
+    directDebitButton = (
+      <DirectDebitPopUpButton
+        callback={postCheckout(
+          props.abParticipations,
+          props.amount,
+          props.csrf,
+          props.currency,
+          props.contributionType,
+          props.country,
+          props.dispatch,
+          'directDebitData',
+          props.referrerAcquisitionData,
+          context.store.getState,
```

Comment:
What's the type of `context`?

Reply:
As discussed offline, this is because we are using the same function that we use in Stripe and PayPal. In those cases, we need the context to get the getState function for postCheckout. Probably in the future, we need to revisit this idea of non-connected shared components.